iavl/iavl_tree.go

```diff
@@ -58,10 +59,21 @@ func (t *IAVLTree) Copy() merkle.Tree {
 		// calculated.
 		t.root.hashWithCount(t)
 	}
-	return &IAVLTree{
+
+	copy := &IAVLTree{
 		root: t.root,
 		ndb:  t.ndb,
 	}
+
+	// Keep track of the trees we create.
+	// Assumes Saves are followed by 2 or less Copies (3 trees)
```

Comment:
Does the fulfillment of these assumptions affect how the library works? I.e. if Save was followed by 4 copied, what would happen?